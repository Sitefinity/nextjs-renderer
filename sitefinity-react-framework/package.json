{
  "name": "sitefinity-react-framework",
  "version": "0.1.0",
  "private": true,
  "publishConfig": {
    "access": "public"
  },
  "license": "See license in EULA.md",
  "dependencies": {
    "@types/react": "18.2.28",
    "react": "18.2.0",
    "tslib": "2.3.1",
    "typescript": "5.2.2"
  },
  "scripts": {
    "build": "tsc"
  }
}
