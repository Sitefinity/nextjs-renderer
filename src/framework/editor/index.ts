export * from './renderer-contract-interfaces';
export * from './services';
export * from './utils';
export * from './widgets';

