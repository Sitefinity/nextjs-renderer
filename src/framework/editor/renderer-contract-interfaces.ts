import { WidgetModel } from "../widgets/widget-model";

export interface RendererContract {
    getWidgetMetadata(args: GetWidgetMetadataArgs): Promise<ComponentMetadata>;
    renderWidget(args: RenderWidgetArgs): Promise<RenderResult>;
    getCategories(args: GetCategoriesArgs): Promise<Array<string>>;
    getWidgets(args: GetWidgetsArgs): Promise<TotalCountResult<Array<WidgetSection>>>;
}

export interface TotalCountResult<T> {
    dataItems: T;
    totalCount: number;
}

export interface RenderResult {
    element: HTMLElement,
    content: string,
    scripts: Array<Script>
}

export interface Script {
    src: string;
    id: string;
}

export interface WidgetItem {
    name: string;
    title?: string;
    addWidgetTitle?: string;
    addWidgetName?: string;
    thumbnailUrl?: string;
    initialProperties?: Array<{ name: string, value: string}>
}

export interface WidgetSection {
    title: string;
    widgets: WidgetItem[];
}

export interface RenderWidgetArgs {
    dataItem: DataItem;
    siteId: string;
    model: WidgetModel<any>;
    token?: Token;
}
export interface GetCategoriesArgs {
    token?: Token;
    toolbox?: string;
}

export interface GetWidgetsArgs {
    dataItem: DataItem;
    category: string;
    search: string;
    skip: number;
    take: number;
    token?: Token;
    toolbox?: string;
}

export interface GetWidgetMetadataArgs {
    dataItem: DataItem;
    widgetKey: string;
    widgetName: string;
    siteId: string;
    token?: Token;
}

export interface ComponentMetadata {
    Caption: string,
    Name: string,
    PropertyMetadata: SectionGroup[],
    PropertyMetadataFlat: PropertyMetadata[],
}

export interface PropertyMetadata {
    Name: string;
    Title: string;
    Type: string;
    DefaultValue: string;
    Properties: { [key: string]: any }
}

export interface SectionGroup {
    Name: string;
    Sections: SectionData[];
}

export interface SectionData {
    Name: string,
    Title: string
    Properties: Array<PropertyMetadata>
}

export interface Token {
    type: string,
    value: string
}

export interface DataItem {
    readonly provider: string;
    readonly culture: string;
    readonly key: string;
}
