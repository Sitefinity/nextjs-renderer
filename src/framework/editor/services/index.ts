export * from './detail-item';
export * from './render-widget-service';
export * from './request-context';
