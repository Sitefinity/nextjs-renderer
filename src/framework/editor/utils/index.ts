export * from './classNames';
export * from './getUniqueId';
export * from './guid';
export * from './rest-extensions';
