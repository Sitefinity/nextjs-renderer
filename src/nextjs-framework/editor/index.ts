export * from './renderer-contract-interfaces';
export * from './utils/classNames';
export * from './utils/getUniqueId';
export * from './utils/guid';

export * from './detail-item';
export * from './request-context';

export * from './widget-framework/attributes';
export * from './widget-framework/content-list-settings';
export * from './widget-framework/link-model';
export * from './widget-framework/mixed-content-context';
export * from './widget-framework/widget-context';
export * from './widget-framework/widget-editor-metadata';
export * from './widget-framework/widget-metadata';
export * from './widget-framework/widget-model';
export * from './widget-framework/widget-registry';
