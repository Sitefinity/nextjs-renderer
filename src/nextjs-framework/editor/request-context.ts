import { PageLayoutServiceResponse, SdkItem } from '../rest-sdk';
import { DetailItem } from './detail-item';

export interface RequestContext {
    layout: PageLayoutServiceResponse;
    searchParams: { [key: string]: string; };
    detailItem?: DetailItem;
    culture: string;
    isEdit: boolean;
    isPreview: boolean;
    isLive: boolean;
}
