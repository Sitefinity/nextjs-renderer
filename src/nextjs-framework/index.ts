export * from './pages/render-page';
export * from './pages/render-widget';
export * from './pages/utils';
