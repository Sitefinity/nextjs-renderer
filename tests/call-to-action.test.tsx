import { CallToAction, CallToActionEntity } from '@progress/sitefinity-react-framework';
import { render, waitFor } from '@testing-library/react';
import { RequestContext, WidgetModel } from '../src/nextjs-framework/editor';

it.skip('should render call to action', async () => {
    const callToAction = await CallToAction({ model, requestContext, metadata });
    const { container } = render(callToAction);

    await waitFor(() => {
        expect(container).toMatchSnapshot();
    });
});

const metadata = {
    designerMetadata: {
        Name: 'SitefinityButton',
        Caption: 'Call to action',
        PropertyMetadata: [Array]
    },
    componentType: [CallToAction],
    editorMetadata: { Title: 'Call to action' },
    ssr: true
};

const model: WidgetModel<CallToActionEntity> = {
    Id: 'c1b4772b-1bd3-4e88-b444-9c6ffa40976e',
    Name: 'SitefinityButton',
    PlaceHolder: 'Column1',
    Caption: 'Call to action',
    ViewName: '',
    Lazy: false,
    Properties: {
        PrimaryActionLink: undefined,
        PrimaryActionLabel: 'Login',
        Margins: undefined,
        SecondaryActionLabel: 'Register',
        SecondaryActionLink: undefined
    },
    Children: []
};

const requestContext: RequestContext = {
    detailItem: null,
    searchParams: {
        sfaction: 'edit',
        sf_culture: 'en',
        sf_site: '4c922118-f076-4e24-9193-93e004f50107',
        sf_page_node: '831706cd-10e1-4184-b2e5-ef9f9ad0fdb5'
    },
    isEdit: true,
    isPreview: false,
    isLive: false,
    culture: 'en',
    pageNode: {
        Culture: 'en',
        SiteId: '4c922118-f076-4e24-9193-93e004f50107',
        MetadataHash: '-1129976431',
        Id: '831706cd-10e1-4184-b2e5-ef9f9ad0fdb5',
        TemplateName: 'Default',
        Renderer: 'React',
        UrlParameters: [],
        HasVariations: false,
        Site: {
            Id: '4c922118-f076-4e24-9193-93e004f50107',
            Name: 'Quantum International',
            DefaultCulture: 'en',
            Cultures: [Array],
            TimeZone: 'UTC;0;UTC;UTC;UTC;;',
            IsSubFolder: false
        },
        User: null,
        ComponentContext: {
            HasLazyComponents: false,
            OrphanedControls: [],
            Components: [Array],
            Fields: null,
            Site: null,
            User: null
        },
        Scripts: [],
        DetailItem: null,
        MetaInfo: {
            Title: 'integration-test',
            Description: '',
            HtmlInHeadTag: null,
            OpenGraphTitle: 'integration-test',
            OpenGraphDescription: '',
            OpenGraphImage: null,
            OpenGraphVideo: null,
            OpenGraphType: null,
            OpenGraphSite: null,
            CanonicalUrl: 'https://localhost:5001/integration-test'
        },
        Fields: {
            Title: 'integration-test',
            UrlName: 'integration-test',
            LastModified: '2023-11-13T13:13:09.183Z',
            DateCreated: '2023-11-13T13:13:09.183Z',
            Crawlable: true,
            ParentId: 'f669d9a7-009d-4d83-ddaa-000000000002',
            RootId: 'f669d9a7-009d-4d83-ddaa-000000000002',
            IsHomePage: false,
            'AvailableLanguages@odata.type': '#Collection(String)',
            AvailableLanguages: [Array],
            ViewUrl: '/integration-test'
        }
    }
};
