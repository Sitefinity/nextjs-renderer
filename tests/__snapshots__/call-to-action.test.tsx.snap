// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`both buttons rendered (Button_Both_Rendered) 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-start"
  >
    <a
      class="me-3 btn btn-primary"
      data-call-to-action=""
      href="kisela-krastavichka.jpg"
    >
      Primary Label
    </a>
    <a
      class="btn btn-secondary"
      data-call-to-action=""
      href="kisela-krastavichka-2.jpg"
    >
      Secondary Label
    </a>
  </div>
</div>
`;

exports[`button attributes rendered (Button_Attributes_Rendered) 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-start"
    wrapper-attribute-key="rapper-attribute-value"
  >
    <a
      class="me-3 btn btn-primary"
      data-call-to-action=""
      href="kisela-krastavichka.jpg"
      primary-attribute-key="primary-attribute-value"
    >
      Primary Label
    </a>
    <a
      class="btn btn-secondary"
      data-call-to-action=""
      href="kisela-krastavichka-2.jpg"
      secondary-attribute-key="secondary-attribute-value"
    >
      Secondary Label
    </a>
  </div>
</div>
`;

exports[`button margins rendered (Button_Margins_Rendered) 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-start mb-4 ms-5"
  >
    <a
      class="me-3 btn btn-primary"
      data-call-to-action=""
      href="kisela-krastavichka.jpg"
    >
      Primary Label
    </a>
  </div>
</div>
`;

exports[`button styling (Button_Styling_Rendered) 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-center"
  >
    <a
      class="me-3 btn btn-secondary"
      data-call-to-action=""
      href="kisela-krastavichka.jpg"
    >
      Primary Label
    </a>
  </div>
</div>
`;

exports[`primary button rendered (Button_PrimaryOnly_Rendered) 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-start"
  >
    <a
      class="me-3 btn btn-primary"
      data-call-to-action=""
      href="kisela-krastavichka.jpg?a=b#c"
    >
      Primary Label
    </a>
  </div>
</div>
`;

exports[`primary class overriden (Button_PrimaryClass_Overriden) 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-start"
  >
    <a
      class="me-3 btn asd"
      data-call-to-action=""
      href="kisela-krastavichka.jpg"
    >
      Primary Label
    </a>
  </div>
</div>
`;

exports[`secondary button rendered (Button_SecondaryOnly_Rendered) 1`] = `
<div>
  <div
    class="d-flex align-items-center justify-content-start"
  >
    <a
      class="btn btn-secondary"
      data-call-to-action=""
      href="kisela-krastavichka.jpg"
    >
      Secondary Label
    </a>
  </div>
</div>
`;
